web/siteplan/src/util/Controls/ExportControl.ts

```diff
@@ -50,182 +70,456 @@ export default class ExportControl extends Control {
-    this.renderToCanvas(pdf, '1')
-    await sleep(5000)
-    pdf.save(this.getExportFileName())
-    this.lockMapDuringExport(false)
+  private getViewportSizePx () {
+    const viewportSize = this.map.getSize()
+    if (!viewportSize) {
+      return undefined
+    }
+
+    const resolution = this.map.getView().getResolution() ?? 1
+    return {
+      width: viewportSize[0] * resolution,
+      height: viewportSize[1] * resolution
+    }
   }
 
-  private getExportFileName (featureData?: SheetCutFeatureData): string {
-    if (featureData?.label && featureData.label.trim().length > 0) {
-      return featureData.label + '.pdf'
+  private createOutsidePolygonFeature (polygon: GeoJSONMultiPolygon|GeoJSONPolygon) {
+    let geometry = null
+    if (polygon.type === 'MultiPolygon') {
+      geometry = new MultiPolygon(polygon.coordinates as GeoJSONPosition[][][])
+    } else {
+      geometry = new Polygon(polygon.coordinates as GeoJSONPosition[][])
     }
 
-    return 'Sl.pdf'
+    const outsidePolygonFeature = new Feature<Geometry>({
+      name: ExportControl.OUTSIDE_POLYGON_FEATURE,
+      geometry
+    })
+
+    outsidePolygonFeature.setStyle(new Style({
+      stroke: undefined
+    }))
+    return outsidePolygonFeature
   }
 
-  private lockMapDuringExport (lock: boolean) {
-    const visibilityState = lock ? 'hidden' : 'visible'
-    // Disable pointer event and hidden all element expect openlayer map by export
-    document.body.style.pointerEvents = lock ? 'none' : 'auto'
-    document.body.style.visibility = visibilityState
-    this.map.getViewport().style.visibility = 'visible'
-    this.map.getOverlayContainerStopEvent().style.visibility = visibilityState
+  private async createTileCanvas (
+    tilesdata: ExportTileData[],
+    visibleLayers: NamedFeatureLayer[],
+    rotationData?: RotateData
+  ) {
+    const viewsize = this.map.getSize()
+    if (!viewsize) {
+      return
+    }
 
-    this.map.getInteractions().forEach(i => i.setActive(!lock))
-    store.commit('setLoading', lock)
+    const horizontalRotate: RotateData | undefined = rotationData ? {
+      rad: rotationData.rad - toRad(90),
+      anchor: rotationData.anchor
+    } : undefined
+    const resolution = this.map.getView().getResolution() as number
+    this.map.getView().setRotation(horizontalRotate ? horizontalRotate.rad : 0)
+    const sleep = (ms: number) => new Promise(res => setTimeout(res, ms))
+    const canvas = document.createElement('canvas')
+    const tileTotalSize = this.getTileTotalSize(tilesdata)
+    canvas.width = tileTotalSize.width
+    canvas.height = tileTotalSize.height
+    const context = canvas?.getContext('2d')
+    if (context == null ){
+      return
+    }
+
+    context.fillStyle = 'white'
+    context.fillRect(0, 0, canvas.width, canvas.height)
+    for (const tileData of tilesdata) {
+      const [tileMinX, tileMinY, tileMaxX, tileMaxY] = tileData.tileExtent
```

Comment:
@TruongQuangSB To avoid linter warnings we could do

```suggestion
      const [tileMinX,,, tileMaxY] = tileData.tileExtent
```